beps/0007-docs-personas-framework-portal/README.md

```diff
@@ -0,0 +1,347 @@
+- Publish a blog post announcing the changes, highlighting Backstage's positioning as both a framework and a developer portal, and explaining the benefits of the restructured documentation.
+
+The benefits of restructuring the documentation according to these ideas include:
+
+1. Easier navigation and discoverability of information for different personas and use cases.
+2. Clear separation of runtime and development documentation.
+3. Simplified process for contributors to determine the appropriate location for new documentation.
+4. Streamlined Backstage adoption process for new adopters.
+
+## Design Details
+
+- The Docs section of the microsite will be divided into two top-level sections: Framework and Developer Portal.
+- The structure of the Table of Contents will align with the outline proposed in https://github.com/backstage/backstage/issues/21946.
+
+### Personas
+
+#### User
+
+Users navigate the developer portal to access tools, information, and plugins essential for their daily tasks. They rely on Backstage to effortlessly find resources, utilize integrations, and connect with other tools and services within their ecosystem. Their interaction is predominantly with the frontend of the portal, where ease of use, accessibility, and relevant content discovery are critical.
+
+##### Documentation Style
+
+Documentation for this persona should be about usability of the portal once it is running. For example:
+
+- Understanding the mechanics of the Software Catalog
+- Registering components
+- Deleting components
+- How the source of truth is the external tool linked through the plugins
+- Understand dependencies relations and the overall schema of the catalog
+- Using available scaffolder actions
+- Customizing new workflows with available actions
+- Searching
+- Using available plugins
+- Step by step tutorials
+
+#### Administrator
+
+Administrators are responsible for the behind-the-scenes technical setup and maintenance of the Backstage portal. This includes deploying the portal, configuring plugins, managing user access, and ensuring the security and performance of the system. They interact with both the frontend and backend, often using command-line tools, administrative dashboards, and configuration files to perform their tasks.
+
+##### Documentation Style
+
+Documentation written for this persona should be DevOps technical, assuming a strong DevOps background. The goal with administrator documentation is to give administrators a strong understanding of how to deploy and manage a Backstage Developer Portal, best practices. For example:
+
+- Installing and upgrading
+- Configuring
+  - Authentication
+  - Plugins
+- Ingesting data (users/groups/components, etc)
+- Installing plugins
+- Implement Git Flows for the Developer portal
+- Creating Pipelines for Docs generation
+- Troubleshooting
+
+#### Integrator
```

Comment:
Tbh I'm not sure the split between admin and integrator is all that useful. My thinking here is that we should merge them and be upfront about the fact that an admin/integrator needs to do some coding to set up Backstage. Over time we strive to remove that, having it become more of "advanced customization" sections where we generally link over to builder docs for the actual instructions instead. What do you think?

Reply:
@Rugvip The distinction between user roles is to help organize the task documentation. The main question should be: What is the relationship between the set of tasks (procedures, processes) that the admin needs to do vs. what the integrator needs to do? If the two sets are largely disjoint, then they should be separate personas. If there's some overlap, it's a judgement call.

My intuition is that the tasks associated with creating a plugin can be self-contained; however, if they're mostly going to be undertaken by admins, then I agree with your arguments for combining the roles. If that's the case, then maybe consider making the the integrator documentation a chapter/section of the admin documentation? That would maintain the separation of the plugin-creation documentation (if I'm creating a plugin, I want all those resources in one place) while making it clear that this is an undertaking for an admin or admin-adjacent persona.

Another advantage to this approach (integration as a subsection of admin) is that it gives you the flexibility to move things around later if/when (as you suggest above) you relieve the admin of some of the coding requirements.

Reply:
@dwelsch-esi It makes sense to me if we define "integrator" and the strict subset of the "admin" persona that needs to do any form of coding. Essentially a persona that we're looking to remove long-term. It also makes sense from a documentation maintenance point of view, since we can eventually just delete that content. My main worry is that it's not going to be a good structure for the docs in the short to medium term, as it might be that the two personas are currently too intertwined.

Plugin creation would be strictly the "builder" persona and I think we should keep that separate from the other two.

Reply:
@Rugvip So are you proposing separating the Builder and Integrator into two different personas? Looking back through the BEP, I see both names are used together ("Integrator/Builder"). My recommendation: If they're the same, pick one name and stick with it. If they're distinct, separate and define them.